PostHog/PostHogSDK.swift

```
@@ -844,21 +844,8 @@ let maxRetryDelay = 30.0
     }
 
     @objc public func isFeatureEnabled(_ key: String) -> Bool {
-        if !isEnabled() {
-            return false
-        }
-
-        guard let remoteConfig else {
-            return false
-        }
-
-        let value = remoteConfig.isFeatureEnabled(key)
-
-        if config.sendFeatureFlagEvent {
-            reportFeatureFlagCalled(flagKey: key, flagValue: value)
-        }
-
-        return value
+        let result = getFeatureFlag(key)
+        return result is String ? true : (result as? Bool) ?? false
```

`getFeatureFlag` returns `Any`, so it can be something different than String and Boo,l and in this ca,se it'd evaluate to `fal,se` which is wrong

---

the correct check would be like this https://github.com/PostHog/posthog-ios/blob/1b011434301250ca75098c40fecad9ba3d209307/PostHog/PostHogRemoteConfig.swift#L384-L393

---

Also since you dont use `isFeatureEnabled` from `PostHogRemoteConfig` anymore, maybe we should remove it?

---

It's too bad Swift doesn't have union types. In practice, a feature flag result will only be `null`, `bool`, or a `string`. If `getFlagValue` returned something other than one of those types, that would be unexpected. In that case, I think returning `false` is appropriate OR we should throw an exception. What do you think?

---

> a feature flag result will only be null, bool, or a string

TIL, so is numbers stringified?
In my opinion, we should never throw an exception; otherwise, clients would always need to wrap their code in try/catch.

---

btw Android does the same so would need fixing https://github.com/PostHog/posthog-android/blob/d269ddb7a1e07590b00d68cd04b62d1b87716554/posthog/src/main/java/com/posthog/PostHog.kt#L737-L765

---

> TIL, so is numbers stringified?

Not really, because there are no numbers for flag results.

We have two types of flags. Boolean and multivariate. A boolean flag result is always a boolean. A variant flag result is the variant key, which is a string.

For example, if you create a multivariate flag, and enter a number as a key, it's stored like so:

```json
{
  "groups": [...],
  "payloads": {},
  "multivariate": {
    "variants": [
      {
        "key": "1234",
        "name": "",
        "rollout_percentage": 0
      }
    ]
  }
}
```

> btw Android does the same so would need fixing [PostHog/posthog-android@d269ddb/posthog/src/main/java/com/posthog/PostHog.kt#L737-L765](https://github.com/PostHog/posthog-android/blob/d269ddb7a1e07590b00d68cd04b62d1b87716554/posthog/src/main/java/com/posthog/PostHog.kt#L737-L765)

Thanks! I was planning to go through every client to see if any others have this issue.

Are you OK with the current changes or is there anything else you'd like to see changed?

---

thanks for the context.
@haacked https://github.com/PostHog/posthog-ios/pull/337#discussion_r2051402773, just this comment, IMO, so we can delete code that isn't used anymore.

---

Oh yeah! I somehow missed that comment. I'll take care of it.

---

Done!